sitepress-rails/lib/sitepress/engine.rb

```diff
@@ -54,11 +54,8 @@ class Engine < ::Rails::Engine
         app.config.cache_classes
       end
 
-      # Setup Sitepress to handle Rails extensions.
-      ActiveSupport.on_load(:action_view) do
-        ActiveSupport.on_load(:after_initialize) do
-          Sitepress::Path.handler_extensions = ActionView::Template::Handlers.extensions
-        end
+      config.to_prepare do
+        Sitepress::Path.handler_extensions = ActionView::Template::Handlers.extensions
```

Comment:
This uses the `config.to_prepare` method call:

https://guides.rubyonrails.org/engines.html#overriding-models-and-controllers

and

https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#autoload-on-boot-and-on-each-reload

![CleanShot 2024-11-23 at 10 06 12@2x](https://github.com/user-attachments/assets/6fe7a212-6083-4382-89d8-4f480812f106)